src/sentry/api/base.py

```diff
@@ -515,6 +515,27 @@ def modify_endpoint_class(self, decorated_class: Type[Endpoint]) -> type:
         new_class.__module__ = decorated_class.__module__
         return new_class
 
+    def create_override(
+        self,
+        original_method: Callable[..., Any],
+        extra_modes: Iterable[SiloMode] = (),
+    ) -> Callable[..., Any]:
+        limiting_override = super().create_override(original_method, extra_modes)
+
+        def single_process_silo_mode_wrapper(*args: Any, **kwargs: Any) -> Any:
+            if SiloMode.single_process_silo_mode():
+                entering_mode: SiloMode = SiloMode.MONOLITH
+                for mode in self.modes:
+                    # Select a mode, if available, from the target modes.
+                    entering_mode = mode
+                with SiloMode.enter_single_process_silo_context(entering_mode):
```

Comment:
`create_override` is a superclass method that creates a function that calls the original, but enforces restricted modes.  I'm adding the additional feature here of 'switching' the current thread's silo context to the endpoint's given silo, assuming that the first entry is the target silo (this is true in practice given `control_silo_endpoint` and `region_silo_endpoint`.)

Notably, endpoints that are either decorated with `pending_silo_endpoint` or no decoration at all will unfortunately default to monolith and won't enforce silo boundaries.  We'll need to tight this up.  In a follow up PR, I may force any registration of endpoints  into django to require having silo endpoint metadata or fail in tests.

Reply:
Yes - let's add this enforcement the same way we did with Models. It might be possible an endpoint needs to be available in both contexts, but that should also be explicit IMO